File: 设计模式/设计模式/适配器模式/Adapter/Adapter/ViewController.m
//
//  ViewController.m
//  Adapter
//
//  Created by yunna on 2019/1/9.
//  Copyright © 2019年 yunna. All rights reserved.
//

#import "ViewController.h"
#import "CardView.h"
#import "OneAdapterModel.h"
#import "TwoAdapterModel.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    CardView *card = [[CardView alloc]initWithFrame:CGRectMake(0, 0, 200, 150)];
    card.center = self.view.center;
    [self.view addSubview:card];
    TwoAdapterModel *model = [[TwoAdapterModel alloc]initWithData:@{@"name":@"辉哥",@"phone":@"19999999999"}];
    [card loadData:model];
    
    
//    OneAdapterModel *model = [[OneAdapterModel alloc]initWithData:@[@"辉哥",@"19999999999"]];
//    [card loadData:model];
    
    
}


@end

File: iOS知识点/iOS大杂烩/image/iOS图片内存优化/imageTest/imageTest/ViewController.swift
//
//  ViewController.swift
//  imageTest
//
//  Created by yunna on 2019/12/20.
//  Copyright © 2019 yunna. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
 
    }


    
    @IBAction func buttonEvent(_ sender: Any) {
        let image = UIImage(named: "shengdanlaoren")
        let iv = UIImageView(frame: CGRect(x: 0, y: 0, width: 100, height: 100))
        iv.image = image
        iv.center = self.view.center
        self.view.addSubview(iv)
    }
    
    
    
}

